api/handler/contact.go

```diff
@@ -39,6 +41,47 @@ func getAllContacts(writer http.ResponseWriter, request *http.Request) {
 	}
 }
 
+func getContactById(writer http.ResponseWriter, request *http.Request) {
+	contactData := request.Context().Value(contactKey).(moira.ContactData)
+	contact, err := controller.GetContactById(database, contactData.ID)
+
+	if err != nil {
+		render.Render(writer, request, err)
+		return
+	}
+
+	if err := render.Render(writer, request, contact); err != nil {
+		render.Render(writer, request, api.ErrorRender(err))
+		return
+	}
+}
+
+func getContactByIdWithEvents(writer http.ResponseWriter, request *http.Request) {
+	contactData := request.Context().Value(contactKey).(moira.ContactData)
+	params := request.URL.Query()
+
+	// TODO: make this checks as the middleware and pass to retrive function
+	if params.Get("from") != "" && params.Get("to") != "" {
```

Comment:
валидацию надо из хендлера в контроллер унести

Reply:
А условие надо инвертировать
```suggestion
	if params.Get("from") == "" || params.Get("to") == "" {
		api.Render(...)
	 	return
	}
```